addons/stock/models/stock_picking.py

```diff
@@ -409,15 +409,16 @@ def _compute_show_check_availability(self):
         """ According to `picking.show_check_availability`, the "check availability" button will be
         displayed in the form view of a picking.
         """
+        rounding = 10 ** -self.env.ref('product.decimal_product_uom').digits
+        StockMove = self.env['stock.move']
         for picking in self:
             if picking.immediate_transfer or not picking.is_locked or picking.state not in ('confirmed', 'waiting', 'assigned'):
                 picking.show_check_availability = False
                 continue
-            picking.show_check_availability = any(
-                move.state in ('waiting', 'confirmed', 'partially_available') and
-                float_compare(move.product_uom_qty, 0, precision_rounding=move.product_uom.rounding)
-                for move in picking.move_lines
-            )
+            picking.show_check_availability = bool(StockMove.search_count(
+                [('picking_id', '=', picking.id),
+                 ('state', 'in', ('waiting', 'confirmed', 'partially_available')),
+                 ('product_uom_qty', '>', rounding)]))
```

Comment:
You should use `('product_uom_qty', '>=', rounding)]))` as your code will not have the same behaviour as float_compare otherwise.

Reply:
There's no simple way to use different rounding for all moves in a search count indeed, but using  `self.env.ref('product.decimal_product_uom')` is such a case is incorrect. The case you linked is very specific because it applies to quants and it's used to perform a SQL Query instead of a search() with the ORM.
Also, if the `product_uom_qty` is the same value than your rounding (thus not null), the check_availability button will be wrongly hidden

Reply:
I have put 0, what do you think ?

Reply:
Problems can occurs with that solution if the system parameter `decimal_product_uom` has a bigger decimal precision than the one on the stock.moves because values could be misinterpreted as 0. But this is a pretty weird usecase. 
@sle-odoo any opinion ?

Reply:
`rounding = 10 ** -(max(6, self.env.ref('product.decimal_product_uom').digits or 0)`